substrate/frame/assets/src/functions.rs

```diff
@@ -438,6 +438,137 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 		Ok(())
 	}
 
+	/// Creates a distribution in storage for asset `id`, which can be claimed via
+	/// `do_claim_distribution`.
+	pub(super) fn do_mint_distribution(
+		id: T::AssetId,
+		merkle_root: DistributionHashOf<T, I>,
+		maybe_check_issuer: Option<T::AccountId>,
+	) -> DispatchResult {
+		let details = Asset::<T, I>::get(&id).ok_or(Error::<T, I>::Unknown)?;
+		ensure!(details.status == AssetStatus::Live, Error::<T, I>::AssetNotLive);
+
+		if let Some(check_issuer) = maybe_check_issuer {
+			ensure!(check_issuer == details.issuer, Error::<T, I>::NoPermission);
+		}
+
+		let info = DistributionInfo {
```

Comment:
Perhaps you have already considered this, but I'm still a little concerned that an account can create a bunch of distributions without paying any storage cost. 

We could limit this by either
- Requiring a deposit when creating a distribution.
- Or, limit it to one active distribution per asset ID.

wdyt?

Reply:
If we don't expect multiple active distribution for an asset, probably that is the most straightforward. We can use AssetId as the key for `MerklizedDistribution`.

Reply:
i am okay with you adding storage deposit to this PR. i would like to hand it off to someone else to own and take to finish line